src/context/TodoContext.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { getTodos, postTodo } from '../api/todos';
+
+type SelectedFilter = 'all' | 'active' | 'completed';
+
+type TodoContext = {
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  error: string | null,
+  handleCloseError: () => void;
+  setNewTodoTitle: (newTodoTitle: string) => void;
+  newTodoTitle: string;
+  setError: (errorName: string) => void;
+  handleSelectFilter: (filterType: SelectedFilter) => void;
+  selectedFilter: SelectedFilter;
+  handleError: (errorName: string) => void;
+  addTodo: (todo: Omit<Todo, 'id'>) => void;
+  USER_ID: number;
+  deleteTodo: (todoId: number) => void;
+  tempTodo: Todo | null;
+  setTempTodo: (todo: Todo | null) => void;
+  disabledInput: boolean;
+  setDisabledInput: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const TodosContext = React.createContext<TodoContext>({
+  todos: [],
+  setTodos: () => { },
+  error: null,
+  handleCloseError: () => { },
+  setNewTodoTitle: () => '',
+  newTodoTitle: '',
+  setError: () => null,
+  handleError: () => null,
+  handleSelectFilter: () => { },
+  selectedFilter: 'all',
+  addTodo: () => { },
+  USER_ID: 0,
+  deleteTodo: () => { },
+  tempTodo: {
+    id: 0, userId: 0, title: '', completed: false,
+  },
+  setTempTodo: () => { },
+  disabledInput: false,
+  setDisabledInput: () => { },
+});
+
+export const TodoProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [selectedFilter, setSelectedFilter] = useState<SelectedFilter>('all');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const USER_ID = 11526;
+
+  const handleError = (errorName: string) => {
+    setError(errorName);
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
```

Comment:
if you make it in useEffect inside component Error, so it would be unnecessary here